test/gtest/common/for_each.cpp

```diff
@@ -45,6 +45,32 @@ TYPED_TEST(ForEach, RangeAlignedZip) {
   EXPECT_EQ(local, dist);
 }
 
+TYPED_TEST(ForEach, LimitedLength) {
+  Ops1<TypeParam> ops(10);
+
+  auto negate = [](auto &&v) { v = -v; };
+  auto input = ops.vec;
+  auto affected_length = rng::size(input) / 2;
+
+  xhp::for_each_n(ops.dist_vec.begin(), affected_length, negate);
+  rng::for_each_n(ops.vec.begin(), affected_length, negate);
+
+  EXPECT_TRUE(check_unary_op(input, ops.vec, ops.dist_vec));
+}
+
+TYPED_TEST(ForEach, WholeLength) {
+  Ops1<TypeParam> ops(10);
+
+  auto negate = [](auto &&v) { v = -v; };
+  auto input = ops.vec;
+  auto affected_length = rng::size(input);
+
+  xhp::for_each_n(ops.dist_vec.begin(), affected_length, negate);
+  rng::for_each(ops.vec.begin(), ops.vec.end(), negate);
+
+  EXPECT_TRUE(check_unary_op(input, ops.vec, ops.dist_vec));
+}
```

Comment:
Please also test with `n` being 0 and `n` exceeding range size.
In order to no duplicate code you may add a helper function `foreach_n_test(uint n)` like sort tests use this [this helper function](https://github.com/oneapi-src/distributed-ranges/blob/main/test/gtest/common/sort.cpp#L12 )